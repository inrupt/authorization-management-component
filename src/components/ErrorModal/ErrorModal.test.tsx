import { render } from "@testing-library/react";
import { ErrorModal } from "./ErrorModal";

describe("ErrorModal", () => {
  it("renders an Error with a title and a message", () => {
    const errorMessage = "This is an error message";
    const errorName = "Error title";
    const e = new Error(errorMessage);
    e.name = errorName;
    const { baseElement } = render(
      <ErrorModal err={e} isOpen={true} onClose={() => {}} />
    );
    expect(baseElement).toMatchSnapshot();
  });

  it("renders an Error with a title, a message and a cause", () => {
    const errorMessage = "This is an error message";
    const errorName = "Error title";
    const e = new Error(errorMessage, { cause: new Error("Error cause") });
    e.name = errorName;
    const { baseElement } = render(
      <ErrorModal err={e} isOpen={true} onClose={() => {}} />
    );
    expect(baseElement).toMatchSnapshot();
  });
});
