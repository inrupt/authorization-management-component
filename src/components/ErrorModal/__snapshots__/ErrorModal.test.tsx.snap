// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorModal renders an Error with a title and a message 1`] = `
<body
  class="modal-open"
  style="overflow: hidden;"
>
  <div />
  <div
    style="position: relative; z-index: 1050; display: block;"
    tabindex="-1"
  >
    <div
      class=""
    >
      <div
        aria-labelledby="modalLabel"
        aria-modal="true"
        class="modal fade"
        role="dialog"
        style="display: block;"
        tabindex="-1"
      >
        <div
          class="modal-dialog confirmation-modal"
          role="document"
        >
          <div
            class="modal-content"
          >
            <div
              class="modal-header modal-header"
            >
              <h5
                class="modal-title"
              >
                <div>
                  <i
                    class="bi bi-exclamation-triangle-fill"
                    title="Confirm"
                  />
                  <h4
                    class="modal-title"
                    id="modalLabel"
                  >
                    Error title
                  </h4>
                </div>
              </h5>
            </div>
            <div
              class="modal-content modal-body"
            >
              <p>
                This is an error message
              </p>
            </div>
            <div
              class="modal-footer modal-footer"
            >
              <button
                class="button primary btn btn-primary modal-primary-btn"
                data-testid="modal-primary-action"
                type="button"
              >
                Ok
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="modal-backdrop fade"
      />
    </div>
  </div>
</body>
`;

exports[`ErrorModal renders an Error with a title, a message and a cause 1`] = `
<body
  class="modal-open"
  style="overflow: hidden;"
>
  <div />
  <div
    style="position: relative; z-index: 1050; display: block;"
    tabindex="-1"
  >
    <div
      class=""
    >
      <div
        aria-labelledby="modalLabel"
        aria-modal="true"
        class="modal fade"
        role="dialog"
        style="display: block;"
        tabindex="-1"
      >
        <div
          class="modal-dialog confirmation-modal"
          role="document"
        >
          <div
            class="modal-content"
          >
            <div
              class="modal-header modal-header"
            >
              <h5
                class="modal-title"
              >
                <div>
                  <i
                    class="bi bi-exclamation-triangle-fill"
                    title="Confirm"
                  />
                  <h4
                    class="modal-title"
                    id="modalLabel"
                  >
                    Error title
                  </h4>
                </div>
              </h5>
            </div>
            <div
              class="modal-content modal-body"
            >
              <details>
                <summary>
                  This is an error message
                </summary>
                Error: Error cause
              </details>
            </div>
            <div
              class="modal-footer modal-footer"
            >
              <button
                class="button primary btn btn-primary modal-primary-btn"
                data-testid="modal-primary-action"
                type="button"
              >
                Ok
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="modal-backdrop fade"
      />
    </div>
  </div>
</body>
`;
